src/model/Types_6D_Modes.h

```diff
@@ -29,37 +32,222 @@ std::ostream& operator<< (std::ostream& s, Result_6D_Modes<REAL> const& args);
 
 #endif
 
+
+
+//template<typename REAL>
+//struct DOF_6D_Modes {
+//	using real_t = typename TypeWrapper<REAL>::real_t;
+//	using vec3_t = Vec3<real_t>;
+//	vec3_t pos;
+//	vec3_t ang;
+//	real_t modes[MODES_LIGAND_MAX_NUMBER];
+//	unsigned int numModes;
+//};
+
+struct Common_Modes {
+	id_t gridLigId;
+	id_t gridRecId;
+	id_t ligId;
+	id_t recId;
+	id_t tableId;
+	id_t paramsId;
+};
+
+//template<typename REAL>
+//class Result_6D_Modes {
+//	using real_t = typename TypeWrapper<REAL>::real_t;
+//	using vec3_t = Vec3<real_t>;
+//	real_t E;
+//	vec3_t pos;
+//	vec3_t ang;
+//	real_t modes[MODES_LIGAND_MAX_NUMBER];
+//	unsigned int numModes;
+//};
+
+
+/**
+ * brief: These types resolve a problem with the types converter in meta.h. Since the modes of DOFs might be varying
+ * (and in multibodydocking even the number of proteins) these types allow to safe their configuration in a simple struct
+ * which is the same for every dof.
+ * This way the solverbase can operate independently of the uses dof type and convert them safely back from Eigen::Vector to the specific DOF
+ * by just setting the DOF configuration.
+ */
+
 template<typename REAL>
-struct DOF_6D_Modes {
+class DOF_6D_Modes {
+public:
 	using real_t = typename TypeWrapper<REAL>::real_t;
 	using vec3_t = Vec3<real_t>;
 	vec3_t pos;
 	vec3_t ang;
 	real_t modes[MODES_LIGAND_MAX_NUMBER];
 	unsigned int numModes;
-};
 
-struct Common_Modes {
-	id_t gridId;
-	id_t ligId;
-	id_t recId;
-	id_t tableId;
-	id_t paramsId;
+
+	TypeConfiguration getConfiguration(){
+		TypeConfiguration config;
+		config._dofSize=1;
+		config._numModes=new int[1];
+		config._numModes[0]=numModes;
+		return config;
+	}
+
+	Vector getVectorfromDOF(){
```

Comment:
You introduce Vector as a dependency here. We should keep it away. This file will also be compiled with CUDA. However, in the GPU-Routines we won't use Vector at all.